matchms/filtering/derive_adduct_from_name.py

```diff
@@ -0,0 +1,48 @@
+from ..typing import SpectrumType
+from ..utils import looks_like_adduct
+
+
+def derive_adduct_from_name(spectrum_in: SpectrumType, remove_adduct_from_name=True) -> SpectrumType:
+    """Find adduct in compound name and add to metadata (if not present yet).
+
+    Method to interpret the given compound name to find the adduct.
+    Args:
+    ----
+    spectrum_in: SpectrumType
+        Input spectrum.
+    remove_adduct_from_name: bool
+        Remove found adducts from compound name if set to True. Default is True.
+    """
+    if spectrum_in is None:
+        return None
+
+    spectrum = spectrum_in.clone()
+
+    # Get compound name
+    if spectrum.get("compound_name", None) is not None:
+        name = spectrum.get("compound_name")
+    else:
+        assert spectrum.get("name", None) in [None, ""], ("Found 'name' but not 'compound_name' in metadata",
+                                                          "Apply 'add_compound_name' filter first.")
+        return spectrum
+
+    # Detect adduct in compound name
+    adduct_from_name = None
+    name_split = name.split(" ")
+    for name_part in name_split[::-1][:2]:
+        if looks_like_adduct(name_part):
+            adduct_from_name = name_part
+            break
+
+    # Remove found adduct from compound name (if remove_adduct_from_name=True)
```

Comment:
The code below is just as clear as the comment. I'd say the comment is unnecessary. If you feel these line need a 'header' or introduction somehow, I'd suggest making it a function (with all its benefits like clear scope and reusability and what not)

Reply:
Comment was removed.